packages/@react-native-windows/cli/src/e2etest/templates/snapshotTemplate.ts

```diff
@@ -0,0 +1,124 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ *
+ * @format
+ */
+
+import glob from 'glob';
+import {promises as fs} from 'fs';
+import path from 'path';
+
+import {copyProjectTemplateAndReplace} from '../../generator-windows';
+import {findPackage} from '@react-native-windows/package-utils';
+import {usingScratchDirectory} from '../sratchDirectory';
+
+type TemplatePermutation = {
+  name: string;
+  experimentalNuGetDependency: boolean;
+  useWinUI3: boolean;
+  useHermes: boolean;
+};
+
+export default function snapshotTemplate(
+  language: 'cpp' | 'cs',
+  projectType: 'app' | 'lib',
+) {
+  // eslint-disable-next-line no-undef
+  console.log = jest.fn();
+
+  const permutations =
+    projectType === 'app'
+      ? generateAppPermutations(language)
+      : generateLibPermutations(language);
+
+  for (const permutation of permutations) {
+    // eslint-disable-next-line no-undef
+    test(permutation.name, async () => {
+      const rnwPackage = await findPackage('react-native-windows');
+      const templateRoot = path.join(rnwPackage!.path, 'template');
+      const name = projectType === 'app' ? 'SnapshotApp' : 'SnapshotLib';
+
+      await usingScratchDirectory(async dir => {
+        await copyProjectTemplateAndReplace(templateRoot, dir, name, name, {
+          projectGuid: '00000000-0000-0000-0000-000000000000',
```

Comment:
nit, should we use a semi-random guid just to catch accidental zero memory guid initialization?
```suggestion
          projectGuid: '77696e64-6f77-7300-0000-0000000FF1CE',
```
wich is windows in hex and office in leet as an homage to: https://docs.microsoft.com/en-us/office/troubleshoot/office-suite-issues/numbering-scheme-for-product-guid

but that is just a bit too cute :) so what about:

```suggestion
          projectGuid: '01234567-0123-0123-0123-0123456789ab',
```